### PR TITLE
Feature/portfolio view flows

### PR DESCRIPTION
**Pull request by antonijzelinskij**
#86966j46j

### CODE REVIEW

File: feature-assets/src/main/java/io/novafoundation/nova/feature_assets/di/AssetsFeatureModule.kt [hunk 1]
```diff
@@ -141,4 +156,20 @@ class AssetsFeatureModule {
         coinPriceRepository = coinPriceRepository,
         poolAccountDerivation = poolAccountDerivation
     )
+
+    @Provides
+    fun provideInitialSwapFlowExecutor(
```
Comment (valentunn) [inline feature-assets/src/main/java/io/novafoundation/nova/feature_assets/di/AssetsFeatureModule.kt:161]: Should this `provideSwapExecutor` and  be `@FeatureScope`?

File: feature-assets/src/main/java/io/novafoundation/nova/feature_assets/domain/networks/AssetNetworksInteractor.kt [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+            .flatMapLatest { walletRepository.syncedAssetsFlow(it.id) }
+
+        return combine(assetsFlow, filterFlow) { assets, filter ->
+            if (filter == null) {
+                assets
+            } else {
+                assets.filter { filter(it) }
+            }
+        }
+    }
```
[the remark this answers is not in the document]
Comment (antonijzelinskij) [reply feature-assets/src/main/java/io/novafoundation/nova/feature_assets/domain/networks/AssetNetworksInteractor.kt:109]: I think this is not critical, this code can change regardless of another interactor

File: feature-assets/src/main/java/io/novafoundation/nova/feature_assets/domain/assets/search/AssetSearchInteractor.kt [hunk 3]
```diff
@@ -75,65 +106,99 @@ class AssetSearchInteractor(
...
+        assetsComparator: Comparator<AssetWithNetwork> = getTokenAssetBaseComparator(),
+        filterFlow: Flow<AssetSearchFilter?>,
+    ): Flow<Map<TokenAssetGroup, List<AssetWithNetwork>>> {
+        val assetsFlow = filteredAssetFlow(filterFlow)
+
+        val aggregatedExternalBalances = externalBalancesFlow.map { it.aggregatedBalanceByAsset() }
+
+        return combine(assetsFlow, aggregatedExternalBalances, queryFlow) { assets, externalBalances, query ->
+            val chainsById = chainRegistry.enabledChainById()
+            val filtered = assets.filterBy(query, chainsById)
```
[the remark this answers is not in the document]
Comment (antonijzelinskij) [reply feature-assets/src/main/java/io/novafoundation/nova/feature_assets/domain/assets/search/AssetSearchInteractor.kt:174]: I found only 2 repeating fragments. I think this is not a problem

File: feature-assets/src/main/java/io/novafoundation/nova/feature_assets/domain/networks/AssetNetworksInteractor.kt [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+            val chainsById = chainRegistry.enabledChainById()
+
+            groupAndSortAssetsByToken(assets, externalBalances, chainsById, assetGroupComparator, assetsComparator)
+                .flatMap { it.value }
+        }
+    }
+
+    private fun filteredAssetFlow(filterFlow: Flow<AssetFilter?>): Flow<List<Asset>> {
+        val assetsFlow = accountRepository.selectedMetaAccountFlow()
+            .flatMapLatest { walletRepository.syncedAssetsFlow(it.id) }
```
[the remark this answers is not in the document]
Comment (antonijzelinskij) [reply feature-assets/src/main/java/io/novafoundation/nova/feature_assets/domain/networks/AssetNetworksInteractor.kt:100]: I agree, however, it doesn't seem to be critical right now

File: feature-assets/src/main/java/io/novafoundation/nova/feature_assets/domain/networks/AssetNetworksInteractor.kt [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+        return combine(assetsFlow, filterFlow) { assets, filter ->
+            if (filter == null) {
+                assets
+            } else {
+                assets.filter { filter(it) }
+            }
+        }
+    }
+
+    private fun getAvailableSwapAssets(tokenSymbol: TokenSymbol, coroutineScope: CoroutineScope): Flow<Set<FullChainAssetId>> {
```
[the remark this answers is not in the document]
Comment (antonijzelinskij) [reply feature-assets/src/main/java/io/novafoundation/nova/feature_assets/domain/networks/AssetNetworksInteractor.kt:111]: Yes, thank you. Fixed that

File: feature-assets/src/main/java/io/novafoundation/nova/feature_assets/presentation/balance/common/baseDecoration/AssetDecorationPreferences.kt [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+import io.novafoundation.nova.common.utils.dp
+import io.novafoundation.nova.feature_assets.presentation.balance.common.TYPE_NETWORK_GROUP
+import io.novafoundation.nova.feature_assets.presentation.balance.common.TYPE_TOKEN_GROUP
+import io.novafoundation.nova.feature_assets.presentation.balance.common.holders.NetworkAssetGroupViewHolder
+import io.novafoundation.nova.feature_assets.presentation.balance.common.holders.NetworkAssetViewHolder
+import io.novafoundation.nova.feature_assets.presentation.balance.common.holders.TokenAssetGroupViewHolder
+
+interface AssetDecorationPreferences {
+
+    fun innerGroupPadding(viewHolder: ViewHolder): Int
```
[the remark this answers is not in the document]
Comment (antonijzelinskij) [reply feature-assets/src/main/java/io/novafoundation/nova/feature_assets/presentation/balance/common/baseDecoration/AssetDecorationPreferences.kt:14]: I think interface is working better and its implementations may be reused in other places that make this code more clean and understandable

File: feature-assets/src/main/java/io/novafoundation/nova/feature_assets/di/AssetsFeatureModule.kt (hunk 1 above)
Comment (antonijzelinskij) [reply feature-assets/src/main/java/io/novafoundation/nova/feature_assets/di/AssetsFeatureModule.kt:161]: Yes, of course. Thank you

**Review by valentunn — APPROVED** [review]

File: feature-assets/src/main/java/io/novafoundation/nova/feature_assets/presentation/balance/common/baseDecoration/AssetDecorationPreferences.kt (hunk 6 above)
[the remark this answers is not in the document]
Comment (valentunn) [reply feature-assets/src/main/java/io/novafoundation/nova/feature_assets/presentation/balance/common/baseDecoration/AssetDecorationPreferences.kt:14]: I dont have anything against using interface here but accepting the ViewHolder here looks overkill. Ideally, functions should accept only those data that is required for them to function. Here. we actually need `Context` and not `ViewModel`. Which can either be passed here or expected to be provided in some other way (e.g. via implementation constructor)

**Pull request merged**